heartbeat/monitors/wrappers/summarizer/summarizer_test.go

```diff
@@ -219,3 +221,181 @@ func TestSummarizer(t *testing.T) {
+
+			// Test locals
+			calls := make(chan string, 100)
+			mtx := sync.Mutex{}
+			appendCall := func(event string) {
+				mtx.Lock()
+				defer mtx.Unlock()
+
+				// Append call to global chan
+				calls <- event
+			}
+
+			// We simplify these to always down since hook order should not be
+			// determined by status
+			job := func(event *beat.Event) (j []jobs.Job, retErr error) {
+
+				calls <- "job"
+
+				event.Fields = mapstr.M{
+					"monitor": mapstr.M{
+						"id":     "test",
+						"status": string(monitorstate.StatusDown),
+					},
+				}
+
+				return nil, fmt.Errorf("dummyerr")
+			}
+
+			s := NewSummarizer(job, sf, tracker)
+			// Shorten retry delay to make tests run faster
+			s.retryDelay = 2 * time.Millisecond
+			// Add mock plugin
+			s.plugins = append(s.plugins, &MockPlugin{
+				eachEvent: func(_ *beat.Event, _ error) {
+					appendCall("ee")
+				},
+				beforeSummary: func(_ *beat.Event) {
+					appendCall("bs")
+				},
+				beforeRetry: func() {
+					appendCall("br")
+				},
+				beforeEachEvent: func(_ *beat.Event) {
+					appendCall("bee")
+				},
+			})
+			wrapped := s.Wrap(job)
+
+			_, _ = jobs.ExecJobAndConts(t, wrapped)
+
+			close(calls)
+
+			// gather order
+			rcvdOrder := []string{}
+			for c := range calls {
+				rcvdOrder = append(rcvdOrder, c)
+			}
+
+			require.Equal(t, tt.expectedOrder, rcvdOrder)
+			require.Len(t, rcvdOrder, len(tt.expectedOrder))
+		})
+	}
+}
+
+func TestRetryLightweightMonitorDuration(t *testing.T) {
+	t.Parallel()
+
+	// Monitor setup
+	tracker := monitorstate.NewTracker(monitorstate.NilStateLoader, false)
+	sf := stdfields.StdMonitorFields{ID: "testmon", Name: "testmon", Type: "http", MaxAttempts: uint16(2)}
+
+	// We simplify these to always down
+	job := func(event *beat.Event) (j []jobs.Job, retErr error) {
+		time.Sleep(1 * time.Second)
+
+		event.Fields = mapstr.M{
+			"monitor": mapstr.M{
+				"id":     "test",
+				"status": string(monitorstate.StatusDown),
+			},
+		}
+
+		return nil, fmt.Errorf("dummyerr")
+	}
+
+	var retryStart time.Time
+
+	s := NewSummarizer(job, sf, tracker)
+	// Shorten retry delay to make tests run faster
+	s.retryDelay = 2 * time.Millisecond
+	// Add mock plugin
+	s.plugins = append(s.plugins, &MockPlugin{
+		beforeRetry: func() {
+			retryStart = time.Now()
+		},
+		eachEvent:       func(_ *beat.Event, _ error) {},
+		beforeSummary:   func(_ *beat.Event) {},
+		beforeEachEvent: func(_ *beat.Event) {},
+	})
+	wrapped := s.Wrap(job)
+
+	events, _ := jobs.ExecJobAndConts(t, wrapped)
+
+	retryElapsed := time.Since(retryStart)
+	require.False(t, retryStart.IsZero())
+	var rcvdDuration interface{}
+	for _, event := range events {
+		summaryIface, _ := event.GetValue("summary")
+		summary := summaryIface.(*jobsummary.JobSummary)
+
+		if summary.FinalAttempt {
+			rcvdDuration, _ = event.GetValue("monitor.duration.us")
+		}
+	}
+	require.Greater(t, rcvdDuration, int64(0))
+	require.GreaterOrEqual(t, look.RTTMS(retryElapsed), rcvdDuration)
```

Comment:
Can you add a comment explaining what this is doing: "Ensures that the delay between attempts is longer than the attempt itself. The retry delay is 1s, but the job should be almost instantaneous"